vignettes/howtogenefilter.Rmd

```diff
@@ -0,0 +1,190 @@
+- name: Khadijah Amusat
+  affiliation: Vignette translation from Sweave to R Markdown / HTML
+date: March 09, 2023
+output:
+  BiocStyle: BiocStyle:: html_documents:
+    number_sections: true
+    toc: true
+    toc_depth: 4
+package: genefilter
+vignette: >
+  %\VignetteIndexEntry{Using the genefilter function to filter genes from a microarray}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Introduction {#introduction .unnumbered}
+
+The *genefilter* package can be used to filter (select) genes from a
+microarray dataset according to a variety of different filtering
+mechanisms. Here, we will consider the example dataset in the
+`sample.ExpressionSet` example from the *Biobase* package. This
+experiment has 26 samples, and there are 500 genes and 3 covariates. The
+covariates are named `sex`, `type` and `score`. The first two have two
+levels and the last one is continuous.
+
+```{r}
+library(Biobase) 
+library(genefilter)
+data(sample.ExpressionSet)
+ varLabels(sample.ExpressionSet)
+ table(sample.ExpressionSet$sex)
+ table(sample.ExpressionSet$type)
+ data(sample.ExpressionSet)
+ 
+```
+
+One dichotomy that can be of interest for subsequent analyses is whether
+the filter is *specific* or *non-specific*. Here, specific means that we
+are filtering with reference to sample metadata, for example, `type`.
+For example, if we want to select genes that are differentially
+expressed in the two groups defined by `type`, that is a specific
+filter. If on the other hand we want to select genes that are expressed
+in more than 5 samples, that is an example of a non--specific filter.
+
+First, let us see how to perform a non--specific filter. Suppose we want
+to select genes that have an expression measure above 200 in at least 5
+samples. To do that we use the function `kOverA`.
+
+There are three steps that must be performed.
+
+1.  Create function(s) implementing the filtering criteria.
+
+2.  Assemble it (them) into a (combined) filtering function.
+
+3.  Apply the filtering function to the expression matrix.
+
+```{r}
+f1 <- kOverA(5, 200)
+ffun <- filterfun(f1)
+wh1 <- genefilter(exprs(sample.ExpressionSet), ffun)
+sum(wh1)
+```
+
+Here `f1` is a function that implies our "expression measure above 200
+in at least 5 samples" criterion, the function `ffun` is the filtering
+function (which in this case consists of only one criterion), and we
+apply it using `genefilter`. There were genes that satisfied the
+criterion and passed the filter.
+As an example for a specific filter, let us select genes that are
+differentially expressed in the groups defined by `type`.
+
+```{r}
+f2 <- ttest(sample.ExpressionSet$type, p=0.1)
+ffun <- filterfun(f1)
+wh2 <- genefilter(exprs(sample.ExpressionSet), 
+filterfun(f2))
+sum(wh2)
+```
+Here, `ttest` is a function from the `genefilter`
+package which provides a suitable wrapper around \texttt{t.test} from
+package \textit{stats}. Now we see that there are  `r \Sexpr{sum(wh2)}`
```

Comment:
Since `stats` is an R package, use `Rpackage` as noted in https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html#4_Style_macros rather than `\textit{}`.

Remove `\Sexpr{}` . I believe this should will still generate the sum of `wh2`.